### PR TITLE
Hide on popper blur

### PR DESCRIPTION
**Pull request by robstax**
# Description
* add hideOnBlur functionality to Popover. when focus moves out of Popover, the popover should close
* added isChildPopoverOpen to allow nested Popovers to be opened within Popovers that have hideOnBlur

# Links
* https://atomiks.github.io/tippyjs/v6/plugins/#hideonpopperblur

### CODE REVIEW

File: src/components/Popover/tippy-plugins/hideOnPopperBlurPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  isChildPopoverOpen?: boolean;
+}
+
+export const hideOnPopperBlurPlugin: Plugin = {
+  name: 'hideOnPopperBlur',
+  defaultValue: true,
+  fn(instance: PopoverInstance & { props: PopperBlurPluginProps }) {
+    return {
+      onCreate() {
+        instance.popper.addEventListener('focusout', (event) => {
```
[the remark this answers is not in the document]
Comment (robstax) [reply src/components/Popover/tippy-plugins/hideOnPopperBlurPlugin.ts:16]: good call out. i remember finding it interesting docs don't have it.

File: src/components/Popover/Popover.tsx [hunk 2]
```diff
@@ -72,6 +74,10 @@ const Popover = forwardRef((props: Props, ref: ForwardedRef<HTMLElement>) => {
     ...rest
   } = props;
 
+  if (hideOnBlur && !disableFocusLock) {
+    console.warn('MRV2 Popover: This component cannot hideOnBlur when focus locked. disableFocusLock must be true if hideOnBlur is true.');
```
[the remark this answers is not in the document]
Comment (robstax) [reply src/components/Popover/Popover.tsx:78]: good catch! will change

File: src/components/Popover/tippy-plugins/hideOnPopperBlurPlugin.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (robstax) [reply src/components/Popover/tippy-plugins/hideOnPopperBlurPlugin.ts:16]: in theory these should be handled by tippyjs and the GC already. see https://github.com/atomiks/tippyjs/blob/ad85f6feb79cf6c5853c43bf1b2a50c4fa98e7a1/src/addons/delegate.ts#L134 and https://github.com/atomiks/tippyjs/issues/824

File: src/components/Popover/Popover.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (nataliadelmar) [reply src/components/Popover/Popover.tsx]: so the condition should be

if (hideOnBlur && !disableFocusLock or hideOnBlur && !interactive) {
      // warn
}

File: src/components/Popover/tippy-plugins/hideOnPopperBlurPlugin.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nataliadelmar) [reply src/components/Popover/tippy-plugins/hideOnPopperBlurPlugin.ts]: is that not for the added listeners on this line https://github.com/atomiks/tippyjs/blob/ad85f6feb79cf6c5853c43bf1b2a50c4fa98e7a1/src/addons/delegate.ts#L104C12-L104C29 ?

[the remark this answers is not in the document]
Comment (robstax) [reply src/components/Popover/tippy-plugins/hideOnPopperBlurPlugin.ts]: cool yep! changed

**Review by nataliadelmar — APPROVED** [review]

**Pull request merged**